cmd/controller/run.go

```diff
@@ -92,6 +93,10 @@ func Run(opts Options, runLog logr.Logger) {
 		os.Exit(1)
 	}
 
+	runLog.Info("setting up metrics")
+	servMetrics := metrics.NewServerMetrics()
+	servMetrics.RegisterAllMetrics()
+
 	// assign bindPort to env var KAPPCTRL_API_PORT if available
```

Comment:
I'd like to help as much as possible,  as this helps me understand kapp-controller a lot better :) but please let me know if you believe you can solve this much faster.

> i do think we should inject an object that holds appropriate fields. seems like having it be a singleton is not a great thing.

Hope this is what you mean